### ISSUE TITLE
Ceph storage for hosting publicly available datasets

### ISSUE THREAD

**Issue by MichaelClifford** [operate-first/odh-moc-support#23]
**Is your feature request related to a problem? Please describe.**

We currently do not have a place where we can publicly host the datasets required to ensure reproducibility of our data science work. Some smaller datasets are currently being stored in github repo's for the time being, but that is only a temporary and non-ideal solution.     

**Describe the solution you'd like**

A ceph bucket that can be used by the operate first team to host relevant datasets. We should be able to open a PR to request read-write credentials as team members. 

There should be a set of publicly available read-only credentials that can be used by anyone to access data for reproducing work in notebooks.    

**Describe alternatives you've considered**
Please see [this issue](https://github.com/aicoe-aiops/data-science-workflows/issues/19) in the ai-coe data science workflow repo where this has been discussed in greater detail. 

But in short these are the ideas we have considered:

* Use a public facing s3 bucket
* Use git lfs within each repo
* Host data on kaggle (similar to some thoth datasets)
* Host data on Backblaze B2

**Comment by chauhankaranraj** [comment]
I think this might be a related issue - https://gitlab.com/open-infrastructure-labs/moc-cnv-sandbox/-/issues/54
@4n4nd could probably confirm

**Comment by 4n4nd** [comment]
Similar request: https://github.com/operate-first/odh-moc-support/issues/18

**Comment by MichaelClifford** [comment]
@4n4nd Can we create a ceph bucket for us Data Science folks in the MOC environment now? :smiley:

**Comment by martinpovolny** [comment]
So this bucket for DS is to be read-write for the data-science group and read-only for the rest of the world.

@tumido, is that correct that currently, such setting requires a manual step as described in https://github.com/operate-first/support/issues/23 ?

**Comment by tumido** [comment]
Yes, deploying policy would require us to do manual steps. I'm also worried you can't have a second set of credentials for a bucket though... I'll ask...

**Comment by martinpovolny** [comment]
Summing up the discussion with @tumido on how to deal with buckets in general so that we can move forward with a proposal on how to solve this issue.

## Individual projects
There's no problem to have one or more buckets provisioned for each project. We already have one for the 'black-flake' project.

## Working with Drif, Hue and possibly other tools

We already have the 'data catalog' component which already has its ArgoCD app and we can add a bucket claim there. Drift and Hue would be configured to have access to this bucket and the bucket would serve as a public "playpen". The credentials would be basically public (pre-configured in Drift and Hue and possibly other places).

## Expose data to the public

@MichaelClifford needs a place where 'data-science' group members can write and anyone (for example running a notebook from JH) can read. This would be used to share examples with the general public.

There should also be a public-facing interface to the storage so that anyone can consume the content of the bucket from outside MOC.

For this, we can use the 'ds-example-project' namespace which already exists. People from the 'data-science' group already have access to this namespace meaning that if we provision a bucket in that namespace they can read the configmap/secret and use the credentials. 

The 'ds-example-project' does not have an ArgoCD app. We would create one so that the bucket claim gets provisioned. Then we would need to manually add the anonymous access permissions as shown here: https://github.com/operate-first/support/issues/23.

**Comment by MichaelClifford** [comment]
@martinpovolny sounds good :smile:

**Comment by tumido** [comment]
FTR, we recently discovered, that there's also a MOC Swift S3 that can be used as a permanent S3 storage facility.

https://github.com/open-infrastructure-labs/ops-issues/issues/33
